addons/hw_drivers/iot_handlers/drivers/L10nKeEDISerialDriver.py

```diff
@@ -0,0 +1,229 @@
+                if response[2] != 0x30 or response[3] != 0x30:
+                    self.data['status'] = response[2:4].decode('cp1251')
+                    _logger.error(
+                        "Sent request: %s,\n Received fiscal device error: %s \n Received command error: %s",
+                        msg.decode('cp1251'), FD_ERRORS.get(response[2], 'Unknown fiscal device error'),
+                        COMMAND_ERRORS.get(response[3], 'Unknown command error'),
+                    )
+                    self.abort_post()
+                    break
+                replies.append('')
+            elif response[0] == 0x15:
+                self.data['status'] = "Received NACK"
+                _logger.error("Sent request: %s,\n Received NACK \x15", msg)
+                self.abort_post()
+                break
+            elif response[0] == 0x02:
+                self.data['status'] = "ok"
+                size = response[1] - 35
+                reply = response[4:4+size]
+                replies.append(reply.decode('cp1251'))
+        return {'replies': replies, 'status': self.data['status']}
+
+    def abort_post(self):
+        """ Cancel the posting of the invoice
+
+        In the event of an error, it is better to try to cancel the posting of
+        the invoice, since the state of the invoice on the device will remain
+        open otherwise, blocking further invoices being sent.
+        """
+        self.message_number += 1
+        abort = struct.pack('BBB', 37, self.message_number+32, 0x39)
+        request = struct.pack('B3s2sB', 2, abort, self.generate_checksum(abort), 10)
+        with self._device_lock:
+            time.sleep(self._protocol.commandDelay)
+            self._connection.write(request)
+            time.sleep(self._protocol.measureDelay)
+            response = self._connection.read_all()
+        if response and response[0] == 0x02:
+            self.data['status'] += "\n The invoice could not be cancelled."
+        else:
+            self.data['status'] += "\n The invoice was successfully cancelled"
+
+
+class TremolG03Controller(http.Controller):
+
+    @http.route('/hw_proxy/l10n_ke_cu_send', type='http', auth='none', cors='*', csrf=False, save_session=False, methods=['POST'])
+    def l10n_ke_cu_send(self, messages, company_vat):
+        """ Posts the messages sent to this endpoint to the fiscal device connected to the server
+
+        :param messages:     The messages (consisting of <CMD> and <DATA>) to
+                             send to the fiscal device.
+        :returns:            Dictionary containing a list of the responses from
+                             fiscal device and status of the fiscal device.
+        """
+        device = TremolG03Driver.get_default_device()
+        if device:
+            # First run the command to get the fiscal device numbers
+            device_numbers = device.send([b'\x60'])
```

Comment:
I guess we do not add this to the regular messages as it does not require abort?